### PR TITLE
`develop` ⬅️ `feature/react-query-volumes`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Final PR to merge `feature/react-query-volumes` ➡️ `develop`
- Moves all Volume data from **Redux** to **React Query**
- Fully API paginates Volumes Landing and Volumes on Linode Detail Page
- Sorry about the commit history, I must have not squashed and merged a PR

## How to test 🧪

- Test all Volume features of Cloud Manager
- If you reviewed #8610, this review should seem similar, we are just merging into `develop` now

### CODE REVIEW

File: packages/manager/src/App.tsx [hunk 1]
```diff
@@ -118,6 +119,13 @@ export class App extends React.Component<CombinedProps, State> {
       .filter((event) => event.action.startsWith('domain') && !event._initial)
       .subscribe(domainEventsHandler);
 
+    /*
+     * Send any Volume events to the Domain events handler in the queries file
```
Comment (dwiley-akamai) [inline packages/manager/src/App.tsx:123]: `to the Volume events handler ...`

File: packages/manager/src/features/Volumes/DestructiveVolumeDialog.tsx [hunk 2]
```diff
@@ -2,108 +2,151 @@ import * as React from 'react';
...
+    delete: 'Delete',
+  }[props.mode];
+
+  const loading = {
+    detach: detachLoading,
+    delete: deleteLoading,
+  }[props.mode];
+
+  const error = {
+    detach: detachError
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Volumes/DestructiveVolumeDialog.tsx:100]: When I block requests to `*volumes*` and try to delete an unattached volume, I still see the detach error 🤔 

Screen Shot 2022-11-28 at 4 41 49 PM

File: packages/manager/src/App.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/App.tsx:123]: Resolved in https://github.com/linode/manager/pull/8629

File: packages/manager/src/features/Volumes/DestructiveVolumeDialog.tsx (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Volumes/DestructiveVolumeDialog.tsx:100]: Resolved in https://github.com/linode/manager/pull/8629

File: packages/manager/src/features/Volumes/VolumeTableRow.tsx [hunk 3]
```diff
@@ -177,13 +198,11 @@ export const VolumeTableRow: React.FC<CombinedProps> = (props) => {
...
         </Grid>
       </TableCell>
-      {isVolumesLanding ? (
-        <TableCell statusCell>
-          <StatusIcon status={volumeStatusIconMap[status]} />
-          {volumeStatusMap[status]}
-        </TableCell>
-      ) : null}
-      {region ? (
+      <TableCell statusCell>
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Volumes/VolumeTableRow.tsx:201]: The mobile view needs some fixing on the Storage tab:

Screen Shot 2022-11-29 at 4 02 56 PM

File: packages/manager/src/features/Volumes/VolumeTableRow.tsx [hunk 4]
```diff
@@ -67,24 +81,46 @@ export const VolumeTableRow: React.FC<CombinedProps> = (props) => {
...
+      event.action === 'volume_migrate' &&
+      event.entity?.id === id &&
+      event.status === 'started'
+  );
+
+  // const recentEvent = events.find((event) => event.entity?.id === id);
+
+  // Use this to show a progress bar
+  // const isUpdating = isVolumeUpdating(recentEvent);
+  // const progress = progressFromEvent(recentEvent);
```
[the remark this answers is not in the document]
Comment (bnussman) [reply packages/manager/src/features/Volumes/VolumeTableRow.tsx:122]: I left this so functionality could be added. I could not find a way to make the progress bar work in prod so I figured id leave this here to be added In the future.

File: packages/manager/src/features/Volumes/VolumeTableRow.tsx (hunk 3 above)
Comment (bnussman) [reply packages/manager/src/features/Volumes/VolumeTableRow.tsx:201]: Fixed in https://github.com/linode/manager/pull/8629

**Comment by mjac0bs** [comment]
@jdamore-linode @bnussman-akamai 

I think I'm seeing some issues with the `attach volume` e2e tests. Usually the first test fails on the first two tries and then seems to usually pass on the third (but not always?). When I've watched it run interactively, I noticed that the Cypress request times out after 30000ms, but it seems like this may be happening due to something go on with a config that is selected, but not recognized as valid in the attach form. I know we made the change to that form's validation as part of the RQ query work. Wasn't sure if that was impacting the tests here or if you were already on top of this, @jdamore-linode 

image

**Comment by jdamore-linode** [comment]
@mjac0bs That's an odd one! Strangely I'm not able to reproduce it -- I'll take a closer look tomorrow morning (and finish reviewing this while I'm at it)

**Review by dwiley-akamai — APPROVED** [review]

**Review by jdamore-linode — APPROVED** [review]

**Pull request merged**